crates/hotshot-stake-table/src/mt_based/config.rs

```diff
@@ -13,9 +13,19 @@ pub(crate) type FieldType = ark_bn254::Fq;
 /// Hash algorithm used in Merkle tree, using a RATE-3 rescue
 pub(crate) type Digest = FixedLengthRescueCRHF<FieldType, TREE_BRANCH, 1>;
 
-/// convert a U256 to a field element.
-pub(crate) fn u256_to_field<F: PrimeField>(v: &U256) -> F {
-    let mut bytes = vec![0u8; 32];
-    v.to_little_endian(&mut bytes);
-    F::from_le_bytes_mod_order(&bytes)
+impl ToFields<FieldType> for FieldType {
+    const SIZE: usize = 1;
+    fn to_fields(&self) -> Vec<FieldType> {
+        vec![FieldType::default()]
+    }
+}
+
+impl ToFields<FieldType> for bls_over_bn254::VerKey {
```

Comment:
Are we also going to implement it for Schnorr verification keys?

Reply:
Check line 334 of vec_based.rs